### PR TITLE
Add Automatic Indentation

### PR DESCRIPTION
**Pull request by will-jac**
This PR is tied to a PR in Ivygate: https://github.com/kipr/ivygate/pull/8

This PR adds automatic indentation to the Simulator, in the form of an indentation button in the editor bar.

The indentation is performed by a custom formatter in Ivygate.

### CODE REVIEW

**Comment by will-jac** [comment]
Also note that the auto-build is failing because it's pointing to a local build of ivygate, for testing before the ivygate PR is merged. After that PR is merged, the version of Ivygate should be changed back

File: src/components/OverlayLayout.tsx [hunk 1]
```diff
@@ -316,6 +316,13 @@ class OverlayLayout extends React.PureComponent<Props, State> {
     let errors = 0;
     let warnings = 0;
 
+    editorBar.push(BarComponent.create(Button, {
+      theme,
+      onClick: onClearConsole,
```
[the remark this answers is not in the document]
Comment (will-jac) [reply src/components/OverlayLayout.tsx:321]: no, definitely not.

File: src/components/OverlayLayout.tsx [hunk 2]
```diff
@@ -316,6 +316,13 @@ class OverlayLayout extends React.PureComponent<Props, State> {
...
+      children:
+        <Fa icon='indent' onClick={() => this.editor_.ivygate.formatCode()}/>
```
[the remark this answers is not in the document]
Comment (will-jac) [reply src/components/OverlayLayout.tsx:323]: hm, I'll see if I can fix that behavior. May be an ivygate/monaco thing

[the remark this answers is not in the document]
Comment (will-jac) [reply src/components/OverlayLayout.tsx:323]: For now I think indent text is fine, although I think hover text would be better

[the remark this answers is not in the document]
Comment (will-jac) [reply src/components/OverlayLayout.tsx]: While it is possible to get and set the cursor position, I couldn't find a fix that felt natural. The problem is that by clicking indent, you're clicking out of the modal, so it loses focus

**Comment by will-jac** [comment]
@navzam I think this should pretty much be ready to go

File: src/components/OverlayLayout.tsx (hunk 2 above)
[the remark this answers is not in the document]
Comment (navzam) [reply src/components/OverlayLayout.tsx]: Gotcha, I think it's really minor so no problem. Can you add an issue for it so we have it tracked?

**Comment by navzam** [comment]
> @navzam I think this should pretty much be ready to go

@will-jac sweet! Can you change the ivygate reference back to github? I merged your ivygate PR and created a `v0.1.1` tag that you should be able to reference

**Comment by will-jac** [comment]
done!

**Comment by navzam** [comment]
> @navzam done! Thanks for your help on this!

@will-jac Hmm I still see the build failing with a bunch of
```
Cannot find module 'ivygate' or its corresponding type declarations.
```

**Comment by will-jac** [comment]
Hm, that was weird, my local build was working even though yarn add ivygate had silently failed. Should be fixed now

**Comment by navzam** [comment]
The linter is failing on one part, but I can address it real quick and then merge!

**Review by navzam — APPROVED** [review]

**Pull request merged**